Linux - Workouts 1 - 3 Improvements (#2096)
enkidevs/curriculum: linux/user-and-file-management/bash-environment/what-are-environment-variables.md

category: must-know

tags:
  - introduction
  - linux
  - environment
  - variables

---

# Environment Variables

---
## Content

An *environment variable* is a variable that affects the way processes behave on a computer. 

> For example, a process might look up the `HOME` environment variable to determine the current user's home directory.

Reviewer: This isn't really correct. An "environment" variable describes its availability, not its capability.

See: https://www.digitalocean.com/community/tutorials/how-to-read-and-set-environmental-and-shell-variables-on-a-linux-vps

Wikipedia has a similar definition but they get away with it because they say "can affect" instead of "affects", which is technically correct but not the point.

```suggestion
An *environment variable* is a variable that is available to all the processes running in the shell.

> 💡You can think of environment variables like global variables

Linux contains many default environment variables.

As an example, the `HOME` environment variable holds the current user's home directory.

> 💡Environment variables are usually named with all uppercase letters to differentiate them from regular variables.
```

Use the `env` command to see a list of all current environment variables:

```bash
env
USER=enki
HOME=/home/enki
# ... (potentially many lines)
```

> You don't have to use `env` to display your environment variables. If you know its name, use the `echo` command to display it:

Reviewer: No need for the first sentence:

```suggestion
If you know the name of the variable you're looking for, use the `echo` command to display it:
```
